lib/screens/change_password.dart

```diff
@@ -0,0 +1,151 @@
+              children: <Widget>[
+                buildNewPasswordField(),
+                buildConfirmPasswordField(),
+                submitButton()
+              ],
+            ),
+          ),
+        );
+      }),
+    );
+  }
+
+  Widget buildNewPasswordField() {
+    return TextFormField(
+      validator: (String value) {
+        if (value.isEmpty) {
+          return 'Please enter password';
+        }
+        if (value.length < 6) {
+          return 'Password should be atleast 6 characters long';
+        }
+      },
+      obscureText: true,
+      controller: _newPasswordTextFieldController,
+      decoration: InputDecoration(
+        labelText: "New Password",
+      ),
+      onSaved: (String value) {
+        setState(() {
+          _newPassword = value;
+        });
+      },
+    );
+  }
+
+  Widget buildConfirmPasswordField() {
+    return TextFormField(
+      validator: (String value) {
+        if (value.isEmpty) {
+          return 'Please confirm new password';
+        }
+        if (_newPasswordTextFieldController.text !=
+            _confirmTextFieldController.text) {
+          return 'Password does not match';
+        }
+      },
+      obscureText: true,
+      controller: _confirmTextFieldController,
+      decoration: InputDecoration(
+        labelText: "Confirm Password",
+      ),
+    );
+  }
+
+  Widget submitButton() {
+    return ScopedModelDescendant(
+        builder: (BuildContext context, Widget child, MainModel model) {
+      return FlatButton(
+          color: Colors.orange,
+          disabledColor: Colors.grey,
+          shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(8)),
+          child: Text(
+            'SET NEW PASSWORD',
+            style: TextStyle(color: Colors.white),
+          ),
+          onPressed: _savingNewPassword
+              ? null
+              : () async {
+                  setNewPassword(context, model);
+                });
+    });
+  }
+
+  setNewPassword(context, model) async {
+    if (!_formKey.currentState.validate()) {
+      return;
+    }
+    setState(() {
+      _savingNewPassword = true;
+    });
+    Map<dynamic, dynamic> updateResponse;
+    _formKey.currentState.save();
+    final SharedPreferences prefs = await SharedPreferences.getInstance();
+    String email = prefs.getString('email');
+    Map<String, String> headers = {'Content-Type': 'application/json'};
+    Map<String, dynamic> payload = Map();
+
+    payload = {
+      'spree_user': {'email': email, 'password': _newPassword}
+    };
+    String url = Settings.SERVER_URL + "auth/change_password";
+
+    http.Response response =
+        await http.put(url, headers: headers, body: json.encode(payload));
+
+    setState(() {
+      _savingNewPassword = false;
+    });
+    if (response.statusCode == 200) {
+      updateResponse = json.decode(response.body);
+      String successMsg = updateResponse['status'];
+      Navigator.popUntil(
```

Comment:
I am not able to view the SnackBar as the Scaffold changes before the snackbar is visible. I think we shouldn't pop the page, or pop it after some delay if needed. Or show an alert instead of a snackbar.  That would be more prominent.